src/Uno.Extensions.Reactive/Collections/Tracking/CollectionAnalyzer.Core.cs

```diff
@@ -0,0 +1,229 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Diagnostics;
+
+namespace Uno.Extensions.Collections.Tracking;
+
+internal partial class CollectionAnalyzer
+{
+	internal static ComparerRef<object?>? GetRef(IEqualityComparer? comparer)
+		=> comparer is null ? null : comparer.Equals;
+
+	internal static ComparerRef<T>? GetRef<T>(IEqualityComparer<T>? comparer)
+		=> comparer is null ? null : comparer.Equals;
+
+	// DEPRECATED: We need to complete extraction of the CollectionUpdater concept
```

Comment:
Link to an existing issue?

Reply:
No GitHub issue, it's more a guidance to what I tend to achieve in term of internal organization of the code